src/components/modules/caret.ts

```diff
@@ -257,15 +258,15 @@ export default class Caret extends Module {
         break;
     }
 
-    /**
-     * @todo try to fix via Promises or use querySelectorAll to not to use timeout
-     */
-    _.delay(() => {
-      this.set(nodeToSet as HTMLElement, offset);
-    }, 20)();
-
     BlockManager.setCurrentBlockByChildNode(block.holder);
     BlockManager.currentBlock.currentInput = element;
+
+    return new Promise<void>((resolve) => {
```

Comment:
why do we need the delay and the Promise?

Reply:
this moment still looks like a not ideal solution. Please, describe the problem you are solving